README.md

```diff
@@ -1,115 +1,624 @@
+$ cat output.detok.txt | sacrebleu ref.detok.txt -m bleu -b -w 4
+20.7965
+```
+
+## Version Signatures
+As you may have noticed, sacreBLEU generates version strings such as `BLEU|nrefs:1|case:mixed|eff:no|tok:13a|smooth:exp|version:2.0.0` for reproducibility reasons. It's strongly recommended to share these signatures in your papers!
+
+## Translationese Support
 
 If you are interested in the translationese effect, you can evaluate BLEU on a subset of sentences
-with a given original language (identified based on the origlang tag in the raw SGM files).
+with a given original language (identified based on the `origlang` tag in the raw SGM files).
 E.g., to evaluate only against originally German sentences translated to English use:
 
-    sacrebleu -t wmt13 -l de-en --origlang=de < my-wmt13-output.txt
+    $ sacrebleu -t wmt13 -l de-en --origlang=de -i my-wmt13-output.txt
 
-and to evaluate against the complement (in this case origlang en, fr, cs, ru, de) use:
+and to evaluate against the complement (in this case `origlang` en, fr, cs, ru, de) use:
 
-    sacrebleu -t wmt13 -l de-en --origlang=non-de < my-wmt13-output.txt
+    $ sacrebleu -t wmt13 -l de-en --origlang=non-de -i my-wmt13-output.txt
 
 *Please note* that the evaluator will return a BLEU score only on the requested subset,
 but it expects that you pass through the entire translated test set.
 
-## Using SacreBLEU from Python
+## Languages & Preprocessing
+
+### BLEU
+
+- You can compute case-insensitive BLEU by passing `--lowercase` to sacreBLEU
+- The default tokenizer for BLEU is `13a` which mimics the `mteval-v13a` script from Moses.
+- Other tokenizers are:
+   - `none` which will not apply any kind of tokenization at all
+   - `char` for language-agnostic character-level tokenization
+   - `intl` applies international tokenization and mimics the `mteval-v14` script from Moses
+   - `zh` separates out **Chinese** characters and tokenizes the non-Chinese parts using `13a` tokenizer
+   - `ja-mecab` tokenizes **Japanese** inputs using the [MeCab](https://pypi.org/project/mecab-python3) morphological analyzer
+- You can switch tokenizers using the `--tokenize` flag of sacreBLEU. Alternatively, if you provide language-pair strings
+  using `--language-pair/-l`, `zh` and `ja-mecab` tokenizers will be used if the target language is `zh` or `ja`, respectively.
+- **Note that** there's no automatic language detection from the hypotheses so you need to make sure that you are correctly
+  selecting the tokenizer for **Japanese** and **Chinese**.
+
+```
+# Default 13a tokenizer will produce poor results for Japanese
+$ sacrebleu kyoto-test.ref.ja -i kyoto-test.hyp.ja -b
+2.1
+
+# Use the ja-mecab tokenizer
+$ sacrebleu kyoto-test.ref.ja -i kyoto-test.hyp.ja --tokenize ja-mecab -b
+14.5
+
+# Alternatively, if you provide the language-pair, sacreBLEU will use ja-mecab automatically
+$ sacrebleu kyoto-test.ref.ja -i kyoto-test.hyp.ja -l en-ja -b
+14.5
+```
+
+### chrF / chrF+
```

Comment:
or chrF++?

Reply:
The naming of chrF+* is a bit weird: Their idea was, for every word n-gram added, you add a `+`. So chrF++ is when `--chrf-word-order == 2`. That's why I decided to refer to the revised version of the metric using `chrF+` in a generic way...

Reply:
OK. Just that chrF++ is mentioned several times above, so the readers/users may get confused.
I would go with
`--chrf-word-order=0` ... chrF
`--chrf-word-order=2` ... chrF++
these are the two common options used in papers. We don't need to introduce a name for a chrF with any other word order.

Thus I suggest also:

> If you enable chrF+ mode by passing `--chrf-word-order` (The number of `+` letters denotes the word n-gram order that you select),
> a very simple punctuation tokenization will be internally applied.

If you enable non-zero `--chrf-word-order`, a very simple punctuation tokenization will be internally applied.

But I don't have any strong opinion on this. Feel free to ignore this suggestion.

Reply:
Makes sense. I'll revise.